core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -205,11 +215,12 @@ private <E extends Exception> boolean runSingleThreaded(
       List<I> succeeded = Lists.newArrayList();
       List<Throwable> exceptions = Lists.newArrayList();
 
-      Iterator<I> iterator = items.iterator();
+      Preconditions.checkState(!hasStarted, "Cannot run twice");
+      hasStarted = true;
```

Comment:
Nit: normally we only use `this.` for field when mutating them. In most cases, this winds up just being the constructor or when set in the builder.

I'm not sure if we would use it here or not.

I wouldn't block on this by any means, but figured I'd mention it 🙂

Reply:
Should i also reference `hasStarted` with `this.` in the checkState above?

should i also reference `items` with `this.` when calling `.next()`? (this is a mutation after all)

Please advise what should i do.

Reply:
Only use `this.` when modifying instance state. It helps the reader know that it is instance state being modified rather than local state. No need to use it when accessing state, only when modifying.

Reply:
Isn’t this line of code modifying instances state though (assigning a class variable from false to true)?